django/contrib/contenttypes/fields.py

```diff
@@ -558,25 +559,37 @@ def get_queryset(self):
                 queryset = super().get_queryset()
                 return self._apply_rel_filters(queryset)
 
+        def get_content_type(self, obj):
+            return ContentType.objects.db_manager(obj._state.db).get_for_model(
+                obj, for_concrete_model=self.for_concrete_model)
```

Comment:
This hook can be added in a separate commit (used also in `GenericRelatedObjectManager.__init__()`).

Reply:
I couldn't understand the logic here. to reach it from other func, Does not have to be a function?  I copied this part from `GenericForeignKey. get_content_type`